src/models/pmv_ppd.js

```diff
@@ -0,0 +1,486 @@
+ * console.log(results.pmv); // Output: [-0.47, 0.06]
+ */
+export function pmv_ppd_array(
+  tdb,
+  tr,
+  vr,
+  rh,
+  met,
+  clo,
+  wme,
+  standard = "ISO",
+  kwargs = {},
+) {
+  const default_kwargs = {
+    units: "SI",
+    limit_inputs: true,
+    airspeed_control: true,
+  };
+  kwargs = Object.assign(default_kwargs, kwargs);
+
+  // default wme is an array of 0
+  if (wme === undefined) {
+    wme = tdb.map(() => 0);
+  }
+
+  if (kwargs.units && kwargs.units === "IP") {
+    // Conversion from IP to SI units
+    const result = units_converter_array({ tdb: tdb, tr: tr, vr: vr }, "IP");
+
+    tdb = result.tdb;
+    tr = result.tr;
+    vr = result.vr;
+  }
+
+  if (standard !== "ISO" && standard !== "ASHRAE") {
+    throw new Error(
+      "PMV calculations can only be performed in compliance with ISO or ASHRAE Standards",
+    );
+  }
+
+  const checkResult = check_standard_compliance_array(standard, {
+    tdb,
+    tr,
+    v: vr,
+    met,
+    clo,
+    airspeed_control: kwargs.airspeed_control,
+  });
+
+  const tdb_valid = checkResult.tdb;
+  const tr_valid = checkResult.tr;
+  const v_valid = checkResult.v;
+  const met_valid = checkResult.met;
+  const clo_valid = checkResult.clo;
+
+  let ce = [];
+  if (standard === "ASHRAE") {
+    ce = vr.map((vrValue, i) => {
+      //if v_r is higher than 0.1 follow methodology ASHRAE Appendix H, H3
+      return vrValue >= 0.1
+        ? cooling_effect(tdb[i], tr[i], vrValue, rh[i], met[i], clo[i], wme[i])
+        : 0;
+    });
+  }
+
+  if (ce.length === 0) {
+    for (let i = 0; i < tdb.length; i++) {
+      ce[i] = 0;
+    }
+  }
+
+  for (let i = 0; i < tdb.length; i++) {
+    tdb[i] -= ce[i];
+    tr[i] -= ce[i];
+    if (ce[i] > 0) {
+      vr[i] = 0.1;
+    }
+  }
+
+  const pmv_array = tdb.map((tdbValue, i) =>
+    pmv_calculation(tdbValue, tr[i], vr[i], rh[i], met[i], clo[i], wme[i]),
+  );
+
+  const ppd_array = pmv_array.map((pmvValue) => {
+    return (
+      100 -
+      95 *
+        Math.exp(
+          -0.03353 * Math.pow(pmvValue, 4) - 0.2179 * Math.pow(pmvValue, 2),
+        )
+    );
+  });
+
+  // Checks that inputs are within the bounds accepted by the model if not return NaN
+  if (kwargs.limit_inputs) {
+    const pmv_valid =
+      standard === "ASHRAE"
+        ? valid_range(pmv_array, [-100, 100])
+        : valid_range(pmv_array, [-2, 2]); // this is the ISO limit
+
+    const all_valid = pmv_array.map((valid, i) => {
+      return !(
+        isNaN(valid) ||
+        isNaN(tdb_valid[i]) ||
+        isNaN(tr_valid[i]) ||
+        isNaN(v_valid[i]) ||
+        isNaN(met_valid[i]) ||
+        isNaN(clo_valid[i]) ||
+        isNaN(pmv_valid[i])
+      );
+    });
+
+    for (let i = 0; i < pmv_array.length; i++) {
+      if (!all_valid[i]) {
+        pmv_array[i] = NaN;
+        ppd_array[i] = NaN;
+      }
+    }
+  }
+
+  return {
+    pmv: pmv_array.map((value) => round(value, 2)),
```

Comment:
can you do this in place rather than creating a new array? (so we avoid allocating more memory)
something like:
```javascript
for (let i = 0; i < pmv_array.length; ++i) {
  pmv_array[i] = round(pmv_array[i], 2)
}
```

Reply:
done